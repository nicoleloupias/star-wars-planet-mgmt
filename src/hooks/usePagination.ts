import { useCallback, useMemo, useState } from "react";

export const usePagination = <T>(data?: T[], size: number = 10) => {
  const totalPages = useMemo(() => {
    if (!data) return 0;

    return Math.ceil(data.length / size);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [size, data?.length]);
  const [page, setPage] = useState(0);

  const onNextPage = useCallback(() => {
    setPage((prevState: number) => {
      if (prevState < totalPages) return prevState + 1;
      return prevState;
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [totalPages]);

  const onPrevPage = useCallback(() => {
    setPage((prevState: number) => {
      if (prevState > 0) return prevState - 1;
      return prevState;
    });
  }, []);

  const slicedData = useMemo(() => {
    if (!data) return data;
    return data.slice(page * size, page * size + size);
  }, [data, size, page]);
  const hasNextPage = useMemo(
    () => !(page === totalPages - 1),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [data, page]
  );
  const hasPrevPage = useMemo(
    () => !(page <= 0),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [data, page]
  );

  return {
    data: slicedData,
    page,
    totalPages,
    setPage,
    onPrevPage,
    onNextPage,
    hasNextPage,
    hasPrevPage
  };
};
