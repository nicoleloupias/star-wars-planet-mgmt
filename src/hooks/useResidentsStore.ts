import { create } from "zustand";
import type { Planet } from "./usePlanetsStore";
import type { Resident } from "../services/types/Resident";

interface ResidentsStore {
  residents: Record<Planet["id"], Resident[]>;
  setResident: (planetId: Planet["id"], resident: Resident) => void;
}

export const useResidentsStore = create<ResidentsStore>((set) => ({
  residents: {},
  setResident: (planetId, resident) => {
    return set((state) => {
      return {
        ...state,
        residents: {
          ...state.residents,
          [planetId]: [...(state.residents?.[planetId] || []), resident]
        }
      };
    });
  }
}));
