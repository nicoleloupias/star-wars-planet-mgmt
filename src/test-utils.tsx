import { ChakraProvider } from "@chakra-ui/react";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import type { MemoryHistory } from "history";
import { createMemoryHistory } from "history";
import React from "react";
import { Router, RouterProvider } from "react-router-dom";
import { theme } from "./theme";
const memoryHistory = createMemoryHistory();

export const AllProviders = ({
  children,
  history = memoryHistory
}: {
  children?: React.ReactNode;
  history?: MemoryHistory;
}) => {
  return (
    <ChakraProvider theme={theme}>
      <Router location={history.location} navigator={history}>
        {children}
      </Router>
    </ChakraProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  {
    options,
    history
  }: {
    options?: RenderOptions;
    history?: MemoryHistory;
  } = {}
): RenderResult =>
  render(ui, {
    wrapper: ({ children }) => <AllProviders history={history}>{children}</AllProviders>,
    ...options
  });

export { customRender as render };
