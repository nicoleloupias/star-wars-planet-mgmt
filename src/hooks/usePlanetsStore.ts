import { create } from "zustand";
import { Planet } from "../services/types/Planet";

interface PlanetStore {
  planets: Planet[];
  setPlanets: (planets: Planet[]) => void;
}

export const usePlanetsStore = create<PlanetStore>((set, get) => ({
  planets: [],
  setPlanets: (planets) => {
    return set((state) => {
      return { ...state, planets: [...state.planets, ...planets] };
    });
  }
}));
