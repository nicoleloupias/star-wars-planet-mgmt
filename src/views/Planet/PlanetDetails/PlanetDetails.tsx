import { Box, Button, Center, Circle, Flex, Heading, ListItem, OrderedList, Text, useToast } from "@chakra-ui/react";
import { Link, useNavigate, useParams } from "react-router-dom";
import { usePlanetsStore } from "../../../hooks/usePlanetsStore";
import stars from "../../../assets/stars.png";
import { useEffect } from "react";
import { getRandomPlanetColor } from "../../../helpers/planet";
import { DeleteIcon, EditIcon } from "@chakra-ui/icons";
import { ResidentService } from "../../../services/ResidentService";
import { useResidentsStore } from "../../../hooks/useResidentsStore";

export const PlanetDetails = () => {
  const { id } = useParams() as { id: string };
  const planets = usePlanetsStore((store) => store.planets);
  const data = planets?.find((planet) => planet.id === id);
  const removePlanet = usePlanetsStore((store) => store.removePlanet);
  const setResident = useResidentsStore((store) => store.setResident);
  const residents = useResidentsStore((store) => store.residents)?.[id];
  const navigate = useNavigate();
  const toast = useToast();
  const planetColor = getRandomPlanetColor(data?.name);

  const handleRemovePlanet = () => {
    const name = data?.name;
    removePlanet(id);
    navigate("/");
    toast({
      status: "success",
      title: `Success! Planet ${name} was removed from our system.`,
      position: "top-right"
    });
  };

  const getResidentId = (url: string) => {
    const regex = /\/people\/(\d+)\//;
    const test = url.match(regex);
    if (!test) return "-1";
    return test[1];
  };

  const getAllResidents = () => {
    data?.residents.forEach(async (residentUrl) => {
      const residentId = getResidentId(residentUrl);
      const resident = await ResidentService.getById(residentId);

      setResident(id, resident);
    });
  };

  useEffect(() => {
    if (!residents) {
      getAllResidents();
    }
  }, [data]);

  return (
    <Flex w="full" minH="100vh" flexDirection={{ base: "column", lg: "row" }} pt={14}>
      <Center bgImage={`url(${stars})`} bgSize="200px" w="full" flex={2 / 3} py={12}>
        <Circle
          size={{ base: "200px", lg: "400px" }}
          bgColor={planetColor}
          boxShadow={`0px 0px 30px 0px ${planetColor}`}
        />
      </Center>
      <Box w="full" flex={1 / 3}>
        <Box bgColor="red.500" p={6} mb={6}>
          <Heading color="gray.50">Planet: {data?.name}</Heading>
        </Box>
        <Box ml={6}>
          <Text>Diameter: {data?.diameter}</Text>
          <Text>Climate: {data?.climate}</Text>
          <Text>Terrain: {data?.terrain}</Text>
          <Text>Population: {data?.population}</Text>

          <Box mt={4}>
            <Heading as="h3" fontSize="xl">
              Residents
            </Heading>

            {residents?.length > 0 ? (
              <OrderedList>{residents?.map(({ name }, i) => <ListItem key={i}>{name}</ListItem>)}</OrderedList>
            ) : (
              <Text>There are no residents in this planet.</Text>
            )}
          </Box>

          <Flex mt={6} gap={4}>
            <Button variant="secondary" to={`/${id}/edit`} as={Link} rightIcon={<EditIcon />}>
              Edit
            </Button>
            <Button
              variant="secondary"
              data-testid="remove-planet-btn"
              onClick={handleRemovePlanet}
              rightIcon={<DeleteIcon />}
            >
              Remove
            </Button>
          </Flex>
        </Box>
      </Box>
    </Flex>
  );
};
