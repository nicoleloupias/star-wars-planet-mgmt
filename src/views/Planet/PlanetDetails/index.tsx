import { Box, Button, Center, Circle, Flex, Heading, Text } from "@chakra-ui/react";
import { Link, useParams } from "react-router-dom";
import { usePlanetsStore } from "../../../hooks/usePlanetsStore";
import stars from "../../../assets/stars.png";
import { useEffect } from "react";
import { getRandomPlanetColor } from "../../../helpers/planet";

export const PlanetDetails = () => {
  const { id } = useParams();
  const data = usePlanetsStore((store) => store.getById)(id || "1");
  const planetColor = getRandomPlanetColor(data?.name);

  const getPlanetDetails = () => {};

  useEffect(() => {
    getPlanetDetails();
  }, []);

  return (
    <Flex w="full" minH="100vh">
      <Center bgImage={`url(${stars})`} bgSize="200px" w="full">
        <Circle size={60} bgColor={planetColor} boxShadow={`0px 0px 30px 0px ${planetColor}`} />
      </Center>
      <Box w="full" bg="gray.700" p={8}>
        <Heading color="gray.50">{data?.name}</Heading>
        <Text>Diameter: {data?.diameter}</Text>
        <Text>Climate: {data?.climate}</Text>
        <Text>Terrain: {data?.terrain}</Text>
        {/* TODO: add residents lists */}
        <Text>Population: {data?.population}</Text>
        <Text>Residents: </Text>

        <Button variant="primary" mt={6} to={`/${id}/edit`} as={Link}>
          Edit
        </Button>
      </Box>
    </Flex>
  );
};
