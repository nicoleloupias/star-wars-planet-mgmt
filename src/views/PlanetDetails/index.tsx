import { Box, Center, Circle, Flex, Heading, Text } from "@chakra-ui/react";
import { useParams } from "react-router-dom";
import { usePlanetsStore } from "../../hooks/usePlanetsStore";
import stars from "../../assets/stars.png";
import { useEffect } from "react";
import { getRandomPlanetColor } from "../../helpers/planet";

export const PlanetDetails = () => {
  const { id } = useParams();
  const data = usePlanetsStore((store) => store.getById)(id || "1");
  const planetColor = getRandomPlanetColor(data?.name);

  const getPlanetDetails = () => {};

  useEffect(() => {
    getPlanetDetails();
  }, []);

  return (
    <Flex w="full" minH="100vh">
      <Center bgImage={`url(${stars})`} bgSize="200px" w="full">
        <Circle size={60} bgColor={planetColor} boxShadow={`0px 0px 30px 0px ${planetColor}`} />
      </Center>
      <Box w="full" bg="gray.700" p={8}>
        <Heading color="gray.50">{data?.name}</Heading>
        <Text>Diameter: {data?.diameter}</Text>
        <Text>Climate: {data?.climate}</Text>
        <Text>Terrain: {data?.terrain}</Text>
        <Text>Population: {data?.population}</Text>
      </Box>
    </Flex>
  );
};
