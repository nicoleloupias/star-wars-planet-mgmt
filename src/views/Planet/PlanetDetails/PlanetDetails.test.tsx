import { beforeEach, describe, expect, test } from "vitest";
import { render } from "../../../test-utils";
import { fireEvent, waitFor } from "@testing-library/react";
import { PlanetDetails } from "./PlanetDetails";
import { usePlanetsStore } from "../../../hooks/usePlanetsStore";
import type { MemoryHistory } from "history";
import { createMemoryHistory } from "history";
import { Route, Routes } from "react-router-dom";
import { useResidentsStore } from "../../../hooks/useResidentsStore";

describe("<PlanetDetails />", () => {
  const planetData = {
    id: "11",
    name: "Geonosis",
    rotation_period: "30",
    orbital_period: "256",
    diameter: "11370",
    climate: "temperate, arid",
    gravity: "0.9 standard",
    terrain: "rock, desert, mountain, barren",
    surface_water: "5",
    population: "100000000000",
    residents: ["https://swapi.dev/api/people/63/"],
    films: ["https://swapi.dev/api/films/5/"],
    created: "2014-12-10T12:47:22.350000Z",
    edited: "2014-12-20T20:58:18.437000Z",
    url: "https://swapi.dev/api/planets/11/"
  };

  let history: MemoryHistory;
  beforeEach(() => {
    history = createMemoryHistory({
      initialEntries: ["/11"]
    });

    usePlanetsStore.setState(
      (state) => ({
        ...state,
        planets: [planetData]
      }),
      true
    );
    useResidentsStore.setState((state) => ({ ...state, residents: {} }), true);
  });

  const factoryComponent = () =>
    render(
      <Routes>
        <Route path="/:id" element={<PlanetDetails />} />
      </Routes>,
      { history }
    );

  test("should render succesfully", () => {
    const component = factoryComponent();

    expect(component).toMatchSnapshot();
  });

  test("should render basic planet data", () => {
    const component = factoryComponent();

    expect(component.getByText(planetData.name, { exact: false })).toBeInTheDocument();
    expect(component.getByText(planetData.diameter, { exact: false })).toBeInTheDocument();
    expect(component.getByText(planetData.population, { exact: false })).toBeInTheDocument();
    expect(component.getByText(planetData.climate, { exact: false })).toBeInTheDocument();
    expect(component.getByText(planetData.terrain, { exact: false })).toBeInTheDocument();
  });

  test("should remove planet when remove btn is clicked", async () => {
    const component = factoryComponent();
    const removePlanetBtn = component.getByRole("button", { name: /Remove/ });
    fireEvent.click(removePlanetBtn);

    expect(history.location.pathname).toBe("/");
    await waitFor(() => {
      expect(component.getByText(`Success! Planet ${planetData.name} was removed from our system.`));
    });
  });

  test("should render a link to edit planet", () => {
    const component = factoryComponent();
    const editPlanetLink = component.getByRole("link", { name: /Edit/ });

    expect(editPlanetLink.getAttribute("href")).toContain("/11/edit");
  });

  test("should render residents", async () => {
    const component = factoryComponent();

    await waitFor(() => {
      expect(component.getByText("Poggle the Lesser - male")).toBeInTheDocument();
    });
  });

  test("should render 'there are no residents' if there are 0 residents", () => {
    usePlanetsStore.setState(
      (state) => ({
        ...state,
        planets: [{ ...planetData, residents: [] }]
      }),
      true
    );
    useResidentsStore.setState({});

    const component = factoryComponent();

    expect(component.getByText(/There are no residents in this planet/));
  });
});
