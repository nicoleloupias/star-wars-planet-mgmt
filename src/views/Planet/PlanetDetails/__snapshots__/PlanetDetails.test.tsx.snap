// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PlanetDetails /> > should render succesfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-dark"
  >
    <div>
      <div
        class="css-1v0rimr"
      >
        <div
          class="css-1wfn8fa"
        >
          <div
            class="css-1grmhqg"
          />
        </div>
        <div
          class="css-15h2xf5"
        >
          <div
            class="css-pg68t7"
          >
            <h2
              class="chakra-heading css-q8veze"
            >
              Planet: 
              Geonosis
            </h2>
          </div>
          <div
            class="css-17n3uhj"
          >
            <p
              class="chakra-text css-0"
            >
              Diameter: 
              11370
               
              <span
                class="css-1nvf5u9"
              >
                km
              </span>
            </p>
            <p
              class="chakra-text css-0"
            >
              Climate: 
              temperate, arid
            </p>
            <p
              class="chakra-text css-0"
            >
              Terrain: 
              rock, desert, mountain, barren
            </p>
            <p
              class="chakra-text css-0"
            >
              Population: 
              100000000000
            </p>
            <div
              class="css-1ayfwcb"
            >
              <h3
                class="chakra-heading css-1llz29a"
              >
                Residents
              </h3>
              <p
                class="chakra-text css-0"
              >
                There are no residents in this planet.
              </p>
            </div>
            <div
              class="css-1kxjbx2"
            >
              <a
                class="chakra-button css-1bvqs64"
                href="/11/edit"
              >
                Edit
                <span
                  class="chakra-button__icon css-1hzyiq5"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-icon css-onkibi"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="none"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-width="2"
                    >
                      <path
                        d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                      />
                      <path
                        d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                      />
                    </g>
                  </svg>
                </span>
              </a>
              <button
                class="chakra-button css-1bvqs64"
                data-testid="remove-planet-btn"
                type="button"
              >
                Remove
                <span
                  class="chakra-button__icon css-1hzyiq5"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-icon css-onkibi"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="currentColor"
                    >
                      <path
                        d="M19.452 7.5H4.547a.5.5 0 00-.5.545l1.287 14.136A2 2 0 007.326 24h9.347a2 2 0 001.992-1.819L19.95 8.045a.5.5 0 00-.129-.382.5.5 0 00-.369-.163zm-9.2 13a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zm5 0a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zM22 4h-4.75a.25.25 0 01-.25-.25V2.5A2.5 2.5 0 0014.5 0h-5A2.5 2.5 0 007 2.5v1.25a.25.25 0 01-.25.25H2a1 1 0 000 2h20a1 1 0 000-2zM9 3.75V2.5a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v1.25a.25.25 0 01-.25.25h-5.5A.25.25 0 019 3.75z"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-label="Notifications-top"
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-top-left"
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-top-right"
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom-left"
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom"
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-bottom-right"
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1v0rimr"
    >
      <div
        class="css-1wfn8fa"
      >
        <div
          class="css-1grmhqg"
        />
      </div>
      <div
        class="css-15h2xf5"
      >
        <div
          class="css-pg68t7"
        >
          <h2
            class="chakra-heading css-q8veze"
          >
            Planet: 
            Geonosis
          </h2>
        </div>
        <div
          class="css-17n3uhj"
        >
          <p
            class="chakra-text css-0"
          >
            Diameter: 
            11370
             
            <span
              class="css-1nvf5u9"
            >
              km
            </span>
          </p>
          <p
            class="chakra-text css-0"
          >
            Climate: 
            temperate, arid
          </p>
          <p
            class="chakra-text css-0"
          >
            Terrain: 
            rock, desert, mountain, barren
          </p>
          <p
            class="chakra-text css-0"
          >
            Population: 
            100000000000
          </p>
          <div
            class="css-1ayfwcb"
          >
            <h3
              class="chakra-heading css-1llz29a"
            >
              Residents
            </h3>
            <p
              class="chakra-text css-0"
            >
              There are no residents in this planet.
            </p>
          </div>
          <div
            class="css-1kxjbx2"
          >
            <a
              class="chakra-button css-1bvqs64"
              href="/11/edit"
            >
              Edit
              <span
                class="chakra-button__icon css-1hzyiq5"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-width="2"
                  >
                    <path
                      d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                    />
                    <path
                      d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                    />
                  </g>
                </svg>
              </span>
            </a>
            <button
              class="chakra-button css-1bvqs64"
              data-testid="remove-planet-btn"
              type="button"
            >
              Remove
              <span
                class="chakra-button__icon css-1hzyiq5"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M19.452 7.5H4.547a.5.5 0 00-.5.545l1.287 14.136A2 2 0 007.326 24h9.347a2 2 0 001.992-1.819L19.95 8.045a.5.5 0 00-.129-.382.5.5 0 00-.369-.163zm-9.2 13a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zm5 0a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zM22 4h-4.75a.25.25 0 01-.25-.25V2.5A2.5 2.5 0 0014.5 0h-5A2.5 2.5 0 007 2.5v1.25a.25.25 0 01-.25.25H2a1 1 0 000 2h20a1 1 0 000-2zM9 3.75V2.5a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v1.25a.25.25 0 01-.25.25h-5.5A.25.25 0 019 3.75z"
                    />
                  </g>
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
